tests/test_xidx_calendar.py

```diff
@@ -0,0 +1,151 @@
+from unittest import TestCase
+
+import pandas as pd
+from pytz import UTC
+from nose_parameterized import parameterized
+
+from .test_trading_calendar import ExchangeCalendarTestBase
+from trading_calendars.trading_calendar import WEEKENDS
+from trading_calendars.exchange_calendar_xidx import XIDXExchangeCalendar
+
+
+class XIDXCalendarTestCase(ExchangeCalendarTestBase, TestCase):
+
+    answer_key_filename = 'xidx'
+    calendar_class = XIDXExchangeCalendar
+
+    MAX_SESSION_HOURS = 6.833
+
+    HAVE_EARLY_CLOSES = False
+
+    # Indonesia Stock Exchange follows the Asia/Jakarta timezone which
+    # does not observe Daylight Saving Time. The timezone offset is
+    # GMT+07:00 for the entire year.
+    DAYLIGHT_SAVINGS_DATES = []
+
+    def test_trading_days(self):
+        # Data obtained from https://www.idx.co.id/en-us/news/trading-holiday/
+        expected_trading_days = {
+            # year: (trading days in year, [trading days in each month])
+            '2016': (246, [20, 20, 21, 21, 20, 22, 16, 22, 21, 21, 22, 20]),
+            '2017': (238, [21, 19, 22, 17, 20, 15, 21, 22, 19, 22, 22, 18]),
+            # XXX: The website says only 21 trading days for July 2018.
+            '2018': (240, [22, 19, 21, 21, 20, 13, 22, 21, 19, 23, 21, 18]),
+            '2019': (245, [22, 19, 20, 19, 21, 15, 23, 22, 21, 23, 21, 19]),
+            '2020': (244, [22, 20, 21, 21, 14, 21, 22, 19, 22, 21, 21, 20]),
+        }
+        for year, (total_days, monthly_days) in expected_trading_days.items():
+            # Sanity checks for the test data.
+            self.assertEqual(len(monthly_days), 12, year)
+            self.assertEqual(sum(monthly_days), total_days, year)
+
+            sessions = self.calendar.sessions_in_range(year, year)
+            self.assertEqual(total_days, len(sessions), year)
```

Comment:
I like this test as a sanity check and verification against the actual stock exchange's website!